test/experimental/main_test.go

```diff
@@ -64,19 +64,21 @@ func TestMain(m *testing.M) {
 	global = environment.NewGlobalEnvironment(ctx)
 
 	// -- Setup the experimental features CM --
-	experimentalFeaturesCm, err := kubeclient.Get(ctx).
+	featuresCm, err := kubeclient.Get(ctx).
 		CoreV1().
 		ConfigMaps(system.Namespace()).
-		Get(ctx, experimental.FlagsConfigName, metav1.GetOptions{})
+		Get(ctx, featureflags.FlagsConfigName, metav1.GetOptions{})
 	if err != nil {
 		panic("Cannot retrieve the experimental features config map")
 	}
 
+	featuresCm.Data[featureflags.KReferenceGroup] = string(featureflags.Allowed)
```

Comment:
This mutates the config map of the cluster without restoring it to the original state, and it is not clear if the cluster can even set that flag on without a restart or wait period. We have no way to know if the configmap change has been accepted by the controllers. 

I would prefer if the test failed if the `featureflags.KReferenceGroup` is not set to `Allowed` over enabling it in the test. 

Cluster mutation should be a cluster setup phase, not done at testing time.

Reply:
Done in another PR (since it's a change related to experimental features per se): https://github.com/knative/eventing/pull/5479